src/Core/SharedKernels/Enumerator/Enumerator/ViewModels/BaseViewModelOfT.cs

```diff
@@ -40,7 +40,14 @@ protected override void SaveStateToBundle(IMvxBundle bundle)
 
         public virtual void Dispose()
         {
+        }
+        
+        public override void ViewDestroy(bool viewFinishing = true)
+        {
+            if(viewFinishing)
+                Dispose();

```

Comment:
possible double dispose

Reply:
Method is virtual. Overrides should handle if it's important for them.